### PR TITLE
[DD4hep] DDFilteredView level and goto methods with unit tests

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* DDFilteredView `level` and `goto` methods with unit tests

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

File: DetectorDescription/DDCMS/src/DDFilteredView.cc [hunk 1]
```diff
@@ -560,6 +560,42 @@ DDFilteredView::nav_type DDFilteredView::navPos() const {
   return pos;
 }
 
+const int DDFilteredView::level() const {
```
Comment (vargasa) [inline DetectorDescription/DDCMS/src/DDFilteredView.cc:563]: @Ianna, Will this conflict with https://github.com/cms-sw/cmssw/pull/31482/files#diff-06d03fa40cda4df4cac28c118cada0eaR563?

File: DetectorDescription/DDCMS/test/BuildFile.xml [hunk 2]
```diff
@@ -4,6 +4,16 @@
   <use name="dd4hep"/>
 </bin>
 
+<bin name="testDD4hepFilteredViewLevel" file="DDFilteredView.level.cppunit.cc,testRunner.cpp">
```
Comment (vargasa) [inline DetectorDescription/DDCMS/test/BuildFile.xml:7]: @ianna I'm a bit confused. Are these `cppunit` files included in other PR?

Comment (vargasa) [reply DetectorDescription/DDCMS/test/BuildFile.xml:7]: I see them here: https://github.com/cms-sw/cmssw/blob/e8a34be27ad271d0e4c56883e9f190d228a99b67/DetectorDescription/DDCMS/test/DDFilteredView.level.cppunit.cc

File: DetectorDescription/DDCMS/src/DDFilteredView.cc (hunk 1 above)
Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc:563]: No worries, I love merging conflicts :-)

File: DetectorDescription/DDCMS/test/BuildFile.xml (hunk 2 above)
Comment (ianna) [reply DetectorDescription/DDCMS/test/BuildFile.xml:7]: Not a problem, I’ll rebase the second in.

**Comment by ianna** [comment]
+1

File: DetectorDescription/DDCMS/src/DDFilteredView.cc [hunk 3]
```diff
@@ -560,6 +560,42 @@ DDFilteredView::nav_type DDFilteredView::navPos() const {
...
+  // try to navigate down to the newpos
+  int level = 1;
+  for (auto const& i : newpos) {
+    it_.back().SetType(0);
+    node = it_.back().Next();
+    for (int j = 1; j <= i; j++) {
+      it_.back().SetType(1);
+      node = it_.back().Next();
+    }
+    level++;
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/src/DDFilteredView.cc:587]: Something is missing here. `level` is incremented but never checked. Or should `level` be deleted?

Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc:587]: Yes, thanks! Well spotted.

Comment (ianna) [reply DetectorDescription/DDCMS/src/DDFilteredView.cc]: removed

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
urgent

**Comment by ianna** [comment]
This PR takes an independent part out of https://github.com/cms-sw/cmssw/pull/31482 to allow a faster integration.

@silviodonato, @dpiparo, @qliphy please, consider integrating it asap. Thanks.

**Comment by silviodonato** [comment]
+1

**Pull request merged**